rspace/src/main/scala/coop/rchain/rspace/spaces/FineGrainedRSpace.scala

```diff
@@ -52,12 +52,17 @@ class FineGrainedRSpace[F[_], C, P, E, A, R, K] private[rspace] (
           logger.error(msg)
           throw new IllegalArgumentException(msg)
         }
+        val span = Kamon.currentSpan()
```

Comment:
cosume mehtod already works with `F[_]`, why not make this part of `Metrics` API and abstrct from using global ?

Reply:
This would require moving the creation of `spans` there as well, and since this is only loosely related I would do that in a separate PR